src/lava/proc/sparse/process.py

```diff
@@ -172,7 +175,10 @@ def __init__(self,
                          **kwargs)
 
         # Transform weights to csr matrix
-        weights = weights.tocsr()
+        if isinstance(weights, np.ndarray):
+            weights = csr_matrix(weights)
+        else:
+            weights = weights.tocsr()
```

Comment:
This code is duplicated. Can this be done only in the `Sparse` base class such that all subclasses pass the weights to it? You do this for `DelaySparse` but not for `LearningSparse`.

Reply:
The difference between DelaySparse and LearningSparse is that LearningSparse primarily inheriting from LearningConnectionProcess. Thus super().init()  of LearningSparse calls init() of LearningConnectionProcess and not of Sparse.